Include/internal/pycore_pystate.h

```diff
@@ -40,6 +34,9 @@ _Py_GetMainConfig(void)
     return _PyInterpreterState_GetConfig(interp);
 }
 
+#ifdef _Py_THREAD_LOCAL
+extern _Py_THREAD_LOCAL PyInterpreterState *_py_current_interpreter;
```

Comment:
Why the current interpreter and not the current `PyThreadState`?

Reply:
Two reasons.
1. I though this would add the most value, as we need the interpreter state to get to freelists which are performance critical.
2. We already pass the threadstate as an argument to a lot of functions, so I wanted to verify this approach before rewriting all that code.

Reply:
Would it make sense to do both?  (They are both hot in different situations.)  What is the cost to the number of thread-local variables?